integration/test/cloudwatchlogs/publish_logs_test.go

```diff
@@ -77,6 +78,39 @@ func TestWriteLogsToCloudWatch(t *testing.T) {
 	}
 }
 
+// Validate https://github.com/aws/amazon-cloudwatch-agent/issues/447
+func TestRotatingLogsDoesNotSkipLines(t *testing.T) {
+	cfgFilePath := "resources/config_log_rotated.json"
+
+	instanceId := test.GetInstanceId()
+	log.Printf("Found instance id %s", instanceId)
+	logGroup := instanceId
+	logStream := instanceId + "Rotated"
+
+	defer test.DeleteLogGroupAndStream(logGroup, logStream)
+
+	start := time.Now()
+	test.CopyFile(cfgFilePath, configOutputPath)
+
+	test.StartAgent(configOutputPath)
+
+	// ensure that there is enough time from the "start" time and the first log line,
+	// so we don't miss it in the GetLogEvents call
+	time.Sleep(agentRuntime)
+	t.Log("Writing logs and rotating")
+	// execute the script used in the repro case
+	test.RunCommand("/usr/bin/python3 resources/write_and_rotate_logs.py")
```

Comment:
Now that the failure is better understood, do you think you could write Golang code instead of calling the python script?
I am fine with keeping the Python script, but I just wanted to check.

Reply:
Tried again briefly and didn't get it right. Pasting the function I wrote for deleting/recreating the log file in case there's something obvious I am missing. I just run this in a loop for the different log lines, and sleep a decent amount of time in between each write. I ran integration tests on my fork and it couldn't even find the log stream, and I don't think I want to invest much more time with this. I think framing it with "something that users will actually use" is good enough of a ~excuse~ justification to me to keep the python script. 
```go
func foo(t *testing.T, filePath, logLine string) {
	os.Remove(filePath) // don't care if it existed before. delete it now
	f, err := os.Create(filePath)
	if err != nil {
		t.Fatalf("Error occurred creating log file for writing: %v", err)
	}
	defer f.Close()

	log.Printf("Writing %s to %s", logLine, filePath)
	_, err = f.WriteString(logLine)
	if err != nil {
		t.Fatalf("Error occurred while writing to %s: %v", filePath, err)
	}
}
```